migration-engine/connectors/sql-migration-connector/src/flavour.rs

```diff
@@ -0,0 +1,253 @@
+}
+
+#[async_trait::async_trait]
+impl SqlFlavour for SqliteFlavour {
+    async fn create_database(&self, database_str: &str) -> ConnectorResult<String> {
+        use anyhow::Context;
+
+        let path = Path::new(&self.file_path);
+        if path.exists() {
+            return Ok(self.file_path.clone());
+        }
+
+        let dir = path.parent();
+
+        if let Some((dir, false)) = dir.map(|dir| (dir, dir.exists())) {
+            std::fs::create_dir_all(dir)
+                .context("Creating SQLite database parent directory.")
+                .map_err(|io_err| ConnectorError::from_kind(migration_connector::ErrorKind::Generic(io_err.into())))?;
+        }
+
+        connect(database_str).await?;
+
+        Ok(self.file_path.clone())
+    }
+
+    async fn describe_schema<'a>(
+        &'a self,
+        schema_name: &'a str,
+        conn: Arc<dyn Queryable + Send + Sync>,
+    ) -> SqlResult<SqlSchema> {
+        Ok(sql_schema_describer::sqlite::SqlSchemaDescriber::new(conn)
+            .describe(schema_name)
+            .await?)
+    }
+
+    async fn initialize(&self, _conn: &dyn Queryable, _database_info: &DatabaseInfo) -> SqlResult<()> {
+        let path_buf = PathBuf::from(&self.file_path);
+        match path_buf.parent() {
+            Some(parent_directory) => {
+                fs::create_dir_all(parent_directory).expect("creating the database folders failed")
+            }
+            None => {}
+        }
+
+        Ok(())
+    }
+}
+
+struct PostgresFlavour(PostgresUrl);
+
+#[async_trait::async_trait]
+impl SqlFlavour for PostgresFlavour {
+    async fn create_database(&self, database_str: &str) -> ConnectorResult<String> {
+        let url = Url::parse(database_str).unwrap();
+        let db_name = self.0.dbname();
+
+        let (conn, _) = create_postgres_admin_conn(url).await?;
+
+        let query = format!("CREATE DATABASE \"{}\"", db_name);
+        catch(
+            conn.connection_info(),
+            conn.query_raw(&query, &[]).map_err(SqlError::from),
+        )
+        .await?;
+
+        Ok(db_name.to_owned())
+    }
+
+    async fn describe_schema<'a>(
+        &'a self,
+        schema_name: &'a str,
+        conn: Arc<dyn Queryable + Send + Sync>,
+    ) -> SqlResult<SqlSchema> {
+        Ok(sql_schema_describer::postgres::SqlSchemaDescriber::new(conn)
+            .describe(schema_name)
+            .await?)
+    }
+
+    async fn initialize(&self, conn: &dyn Queryable, database_info: &DatabaseInfo) -> SqlResult<()> {
+        let schema_sql = format!(
+            "CREATE SCHEMA IF NOT EXISTS \"{}\";",
+            &database_info.connection_info().schema_name()
+        );
+
+        conn.query_raw(&schema_sql, &[]).await?;
+
+        Ok(())
+    }
+}
+
+/// Try to connect as an admin to a postgres database. We try to pick a default database from which
+/// we can create another database.
+async fn create_postgres_admin_conn(mut url: Url) -> ConnectorResult<(Quaint, DatabaseInfo)> {
```

Comment:
This is just moved.

Reply:
Could you maybe split the database-specific stuff to their own modules?